fix/workaround ROCm 5.2.0 compiler bug

psychocoderHPC opened the pull request:
fix #4305

Fix broken `KernelShiftParticles` most likely produced by a compiler bug.

The workaround is not showing adverse effects on the registered footprint on NVIDIA GPUs (tested sm_70).

PrometheusPi commented on test/KHI_growthRate/include/picongpu/param/grid.param:
```diff
@@ -45,7 +45,6 @@ namespace picongpu
 {
     namespace SI
     {
-
```
Not related to this PR - but fine with me

psychocoderHPC replied:
Ohh, this should not be changed.

psychocoderHPC committed:
fix/workaround ROCm 5.2.0 compiler bug

fix #4305

Fix broken KernelShiftParticles most likely produced by a compiler bug.

psychocoderHPC commented:
@PrometheusPi I removed the unnecessary change, the PR can be merged.
